subprojects/docs/src/samples/userguide/tasks/incrementalTask/kotlin/build.gradle.kts

```diff
@@ -0,0 +1,81 @@
+task("originalInputs") {
+    doLast {
+        file("inputs").mkdir()
+        file("inputs/1.txt").writeText("Content for file 1.")
+        file("inputs/2.txt").writeText("Content for file 2.")
+        file("inputs/3.txt").writeText("Content for file 3.")
+    }
+}
+
+// tag::updated-inputs[]
+task("updateInputs") {
+    doLast {
+        file("inputs/1.txt").writeText("Changed content for existing file 1.")
+        file("inputs/4.txt").writeText("Content for new file 4.")
+    }
+}
+// end::updated-inputs[]
+
+// tag::removed-input[]
+task("removeInput") {
+    doLast {
+        file("inputs/3.txt").delete()
+    }
+}
+// end::removed-input[]
+
+// tag::removed-output[]
+task("removeOutput") {
+    doLast {
+        file("$buildDir/outputs/1.txt").delete()
+    }
+}
+// end::removed-output[]
+
+// tag::reverse[]
+task<IncrementalReverseTask>("incrementalReverse") {
+    inputDir = file("inputs")
+    outputDir = file("$buildDir/outputs")
+    inputProperty = project.properties["taskInputProperty"] as String? ?: "original"
+}
+// end::reverse[]
+
+// tag::incremental-task[]
+open class IncrementalReverseTask : DefaultTask() {
+    @InputDirectory
+    var inputDir: File? = null
```

Comment:
:x: Let's make this `lateinit var inputDir: File` instead. We don't expect any null value for this property, it's not _optional_.